azurelinuxagent/ga/cgroupconfigurator.py

```diff
@@ -150,39 +139,58 @@ def initialize(self):
+                    # the kernel related to cgroups.
+                    log_cgroup_info("Agent will reset the quotas in case distro: {0} went from supported to unsupported".format(get_distro()), send_event=False)
+                    self._reset_agent_cgroup_setup()
                     return
 
                 # check that systemd is detected correctly
-                self._cgroups_api = SystemdCgroupsApi()
                 if not systemd.is_systemd():
-                    _log_cgroup_warning("systemd was not detected on {0}", get_distro())
+                    log_cgroup_warning("systemd was not detected on {0}".format(get_distro()))
                     return
 
-                _log_cgroup_info("systemd version: {0}", systemd.get_version())
+                log_cgroup_info("systemd version: {0}".format(systemd.get_version()))
+
+                # Determine which version of the Cgroup Api should be used. If the correct version can't be determined,
+                # do not enable resource monitoring/enforcement.
+                try:
+                    self._cgroups_api = get_cgroup_api()
+                except InvalidCgroupMountpointException as e:
+                    # Systemd mounts the cgroup file system at '/sys/fs/cgroup'. Previously, the agent supported cgroup
+                    # usage if a user mounted the cgroup filesystem elsewhere. The agent no longer supports that
+                    # scenario. Cleanup any existing drop in files in case the agent previously supported cgroups on
+                    # this machine.
+                    log_cgroup_warning("The agent does not support cgroups if the default systemd mountpoint is not being used: {0}".format(ustr(e)), send_event=True)
+                    log_cgroup_info("Agent will reset the quotas in case cgroup usage went from enabled to disabled")
+                    self._reset_agent_cgroup_setup()
+                    return
+                except CGroupsException as e:
+                    log_cgroup_warning("Unable to determine which cgroup version to use: {0}".format(ustr(e)), send_event=True)
+                    return
 
                 if not self.__check_no_legacy_cgroups():
                     return
 
                 agent_unit_name = systemd.get_agent_unit_name()
                 agent_slice = systemd.get_unit_property(agent_unit_name, "Slice")
                 if agent_slice not in (AZURE_SLICE, "system.slice"):
-                    _log_cgroup_warning("The agent is within an unexpected slice: {0}", agent_slice)
+                    log_cgroup_warning("The agent is within an unexpected slice: {0}".format(agent_slice))
                     return
 
                 self.__setup_azure_slice()
 
-                cpu_controller_root, memory_controller_root = self.__get_cgroup_controllers()
-                self._agent_cpu_cgroup_path, self._agent_memory_cgroup_path = self.__get_agent_cgroups(agent_slice,
+                if self.cgroup_v2_enabled():
```

Comment:
`BTW with this PR, the azure slice/drop in files are not created until after we check if v2 is in use. So on new machines with this update, drop-in files won't be created if v2 is in use.`

@maddieford  you made this comments, do you have a plans to change this condition and place above `self.__setup_azure_slice()` As of now it's not matching what you said.

Reply:
You're right, I thought I moved it. I'll move it in next commit